modules/ingestor/src/service/advisory/csaf/loader.rs

```diff
@@ -51,18 +51,15 @@ impl<'g> CsafLoader<'g> {
         Self { graph }
     }
 
-    #[instrument(skip(self, labels, document), err)]
-    pub async fn load<R: Read>(
+    #[instrument(skip(self, labels), err)]
+    pub async fn load(
         &self,
         labels: impl Into<Labels>,
-        document: R,
+        csaf: Csaf,
```

Comment:
I know WIP, but... would consuming the `Read` not also consume/drop the underlying buffer?  Or does `serde` maintain references into the original bytes? Maybe deserialize-owned?

Reply:
I think you're correct, but it's not the specific problem I'm attempting to solve. What I'm trying to avoid is the `SyncAdapter`. That's the thing loading the entire doc into memory. I'm hoping that using the `SyncIoBridge` instead will minimize memory usage at the cost of spawning a thread.

Ultimately, our troubles come down to serde not supporting async. And I'm trying to come up with a `Read` impl that doesn't blow out our RAM.

I'm open to other ideas, of course.